torch/csrc/distributed/autograd/engine/dist_engine.cpp

```diff
@@ -180,6 +218,22 @@ void DistEngine::computeDependencies(
     // Create a dummy GraphRoot and run init_to_execute with it.
     GraphRoot dummyRoot(edges, {});
     graphTask->init_to_execute(dummyRoot, outputEdges);
+    for (auto& pr : graphTask->exec_info_) {
+      auto fn = pr.first;
+      auto& execInfo = pr.second;
+      if (!execInfo.captures_) {
+        continue;
+      }
+      if (auto accumulateGradFn = dynamic_cast<AccumulateGrad*>(fn)) {
```

Comment:
Shouldn't this be an assert? 
This whole logic is going to do weird things if the node is not an accumulate grad (namely not capture the gradients properly).

Reply:
Wil do that.